scripts/bump_version.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Get the path of the script
+SCRIPT_PATH=$(dirname $(realpath -s $0))
+
+####################
+# Check preconditions
+####################
+
+# Check for a version number passed by argument
+if [ "$#" -ne 1 ]; then
+    echo "Usage: $0 version-number"
+    exit 1
+fi
+
+# Validate version - source this so we get the version information
+source ${SCRIPT_PATH}/validate_version.sh $1
+if [ $? -ne 0 ]; then
+    exit 1
+fi
+
+####################
+# Bump version
+####################
+
+# Update version in the CMakeLists.txt
+sed -i "s/set (CPACK_PACKAGE_VERSION_MAJOR \"[0-9]\+\")/set (CPACK_PACKAGE_VERSION_MAJOR \"$MAJOR_VERSION\")/g" $SCRIPT_PATH/../CMakeLists.txt
+sed -i "s/set (CPACK_PACKAGE_VERSION_MINOR \"[0-9]\+\")/set (CPACK_PACKAGE_VERSION_MINOR \"$MINOR_VERSION\")/g" $SCRIPT_PATH/../CMakeLists.txt
+sed -i "s/set (CPACK_PACKAGE_VERSION_PATCH \"[0-9]\+\")/set (CPACK_PACKAGE_VERSION_PATCH \"$PATCH_VERSION\")/g" $SCRIPT_PATH/../CMakeLists.txt
+sed -i "s/set (VERSION_SUFFIX \"[^\"]*\")/set (VERSION_SUFFIX \"$VERSION_SUFFIX\")/g" ${SCRIPT_PATH}/../CMakeLists.txt
+
+# Update Changelog
+if ! grep -Fxq "## $1" $SCRIPT_PATH/../CHANGELOG.md; then
+    sed -i "N;N;s/# Changelog\n\n## Nightly (Unreleased)/# Changelog\n\n## Nightly (Unreleased)\n\n## $VERSION\n\nTODO/g" $SCRIPT_PATH/../CHANGELOG.md
+    echo "WARNING: Please make sure to update CHANGELOG.md"
+else
+    echo "ERROR: CHANGELOG.md already contains this version number"
```

Comment:
The way I understand it:
* When publishing a release, it may not be listed in the changelog, so we emit a warning.
* If we publish a release but it is already in the changelog, it has likely to have already been published. So raise an error.

I think it might make more sense to make both of them errors, so that a release _must_ be accompanied by a changelog update.

Reply:
It might also make sense to `exit 1` at the end of this script if we encounter an error. E.g.

```sh
EXIT_STATUS=0

displayError() {
    echo "ERROR: $1"
    EXIT_STATUS=1
}

...

else
    displayError "Some error"
fi

exit $EXIT_STATUS
```

> I think it might make more sense to make both of them errors, so that a release must be accompanied by a changelog update.

I think this makes sense. In this case, however, the above suggestion (`exit $EXIT_STATUS`) could be replaced with `exit 1` and wouldn't be particularly useful.

Reply:
I do not see the changelog as a source of truth. Instead the major scripts use git tags and/or CMake to validate the version string. 

The reasoning behind this is, that the changelog must be updated manually and therefore somebody might choose to do this prematurely. In either case, a manual check of the changelog is required in all scenarios, so this should be a warning after all and no error.